projects/pastanaga-angular/src/lib/controls/toggles/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,108 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    Self,
+    SimpleChanges,
+    ViewChild
+} from '@angular/core';
+import { BaseControl } from '../../base-control';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { detectChanges } from '../../../common';
+import { NgControl, ValidatorFn } from '@angular/forms';
+
+@Component({
+    selector: 'pa-checkbox',
+    templateUrl: './checkbox.component.html',
+    styleUrls: ['./checkbox.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CheckboxComponent extends BaseControl implements OnInit, OnDestroy {
+    @Input() set type(value: 'checkbox' | 'radio') {
+        this._type = value;
+        // When using Angular inputs, developers are no longer able to set the properties on the native
+        // input element. To ensure that bindings for `type` work, we need to sync the setter
+        // with the native property.
+        if (!!this.htmlElement) {
+            this.htmlElement.type = value;
+        }
+    }
+
+    get type(): 'checkbox' | 'radio' {
+        return this._type;
+    }
+
+    @Input() required?: boolean;
+
+    @ViewChild('htmlElement') htmlElementRef?: ElementRef;
+
+    @Output() selectedChange: EventEmitter<boolean> = new EventEmitter();
+
+    _type: 'checkbox' | 'radio' = 'checkbox';
+    isChecked = false;
+
+    requiredValidator?: ValidatorFn;
+
+    constructor(@Optional() @Self() public parentControl: NgControl, protected cdr: ChangeDetectorRef) {
+        super(parentControl, cdr);
+        this._fieldKind = this._type;
+    }
+
+    ngOnInit(): void {
+        super.ngOnInit();
+    }
+
+    ngOnDestroy() {
+        super.ngOnDestroy();
+    }
+
+    toggleCheckbox() {
+        if (this._type === 'radio' && !this.model) {
+            this.isChecked = true;
+            this.onChange(true);
+        } else if (this._type === 'checkbox' || !this.model) {
+            this.isChecked = !this.isChecked;
+            this.onChange(this.isChecked);
```

Comment:
maybe we should add a comment above this line... just to explain that onChange won't do anything when the component is managed by ControlValueAccessor (ngModel or FormControl)... i don't know...